backend/new_app.py

```diff
@@ -0,0 +1,408 @@
+        Role_Description=role_description,
+        Role_Requirements=role_requirements
+    )
+
+    try:
+        db.session.add(new_role)
+        db.session.commit()
+    except Exception as e:
+        db.session.rollback()
+        return jsonify({
+            'error': str(e)
+            }
+        ), 500
+
+    return jsonify(
+        {
+            "code": 201,
+            "data": new_role.json(),
+            "message": "Role listing created successfully"
+        }
+    ), 201
+
+# for HR to update role
+@app.route('/role/update/<int:role_id>', methods=['PUT'])
+def update_role(role_id):
+    role = Role.query.get(role_id)
+    if not role:
+        return jsonify({'message': 'Role not found'}), 404
+
+    # link data to frontend to pass info
+    data = request.get_json()
+    role.Role_Name = data['Role_Name']
+    role.Date_Posted = data['Date_Posted']
+    role.App_Deadline = data['App_Deadline']
+    role.Role_Department = data['Role_Department']
+    role.Role_Description = data['Role_Description']
+    role.Role_Requirements = data['Role_Requirements']
+
+    skills = data.get('Role_Skills', [])  # assuming Role_Skills is a list of skill names
+    # clear existing role skills
+    Role_Skill.query.filter_by(Role_Name=role.Role_Name).delete()
+    # add updated role skills
+    for skill_name in skills:
+        role_skill = Role_Skill(Role_Name=role.Role_Name, Skill_Name=skill_name)
+        db.session.add(role_skill)
+
+    db.session.commit()
+
+    return jsonify({'message': 'Role updated successfully'}), 200
+
+
+# get skills required for the role
+@app.route('/roles/<string:role_name>/skills', methods=['GET'])
+def get_role_skills(role_name):
+    role_skills = Role_Skill.query.filter_by(Role_Name=role_name).all()
+    if not role_skills:
+        return jsonify({'message': 'Role not found'}), 404
+
+    results = [role_skill.json() for role_skill in role_skills]
+
+    return jsonify(results)
+
+
+
+# staff endpoints
+
+# to create new staff
+@app.route('/staff/create', methods=['POST'])
+def create_staff():
+    data = request.get_json()
+
+    staff_fname = data['Staff_FName']
+    staff_lname = data['Staff_LName']
+    dept = data['Dept']
+    country = data['Country']
+    email = data['Email']
+    access_rights = data['Access_Rights']
+
+    # Check if staff member already exists by email
+    if Staff.query.filter_by(Email=email).first():
+        return jsonify(
+            {
+                "code": 400,
+                "message": "Staff member with this email already exists!"
+            }
+        ), 400
+
+    # If staff member doesn't exist, insert into the database
+    new_staff = Staff(
+        Staff_FName=staff_fname,
+        Staff_LName=staff_lname,
+        Dept=dept,
+        Country=country,
+        Email=email,
+        Access_Rights=access_rights
+    )
+    
+    try:
+        db.session.add(new_staff)
+        db.session.commit()
+    except Exception as e:
```

Comment:
same here, would be good to specify the IntegrityError instead of a general exception